File: pkg/codegen/testing/test/testdata/unknown-resource-pp/python/unknown-resource.py
import pulumi
import pulumi_unknown as unknown

provider = pulumi.providers.Unknown("provider")
main = unknown.index.Main("main",
    first=hello,
    second={
        foo: bar,
    })
from_module = []
for range in [{"value": i} for i in range(0, 10)]:
    from_module.append(unknown.eks.Example(f"fromModule-{range['value']}", associated_main=main.id))
pulumi.export("mainId", main["id"])
pulumi.export("values", from_module["values"]["first"])

File: pkg/codegen/testing/utils/testdata/unknown-resource-pp/dotnet/unknown-resource.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Unknown = Pulumi.Unknown;

return await Deployment.RunAsync(() => 
{
    var provider = new Pulumi.Providers.Unknown("provider");

    var main = new Unknown.Index.Main("main", new()
    {
        First = "hello",
        Second = 
        {
            { "foo", "bar" },
        },
    });

    var fromModule = new List<Unknown.Eks.Example>();
    for (var rangeIndex = 0; rangeIndex < 10; rangeIndex++)
    {
        var range = new { Value = rangeIndex };
        fromModule.Add(new Unknown.Eks.Example($"fromModule-{range.Value}", new()
        {
            AssociatedMain = main.Id,
        }));
    }
    return new Dictionary<string, object?>
    {
        ["mainId"] = main.Id,
        ["values"] = fromModule.Values.First,
    };
});

